File: webapp/src/app/interfaces/item.ts
import type { Movement } from './movement';
import type { User } from './user';

export interface Item {
  id: number;
  category: string;
  name: string;
  brand: string;
  model: string;
  serialNumber: string;
  patrimonyNumber: string;
  description: string;
  purchaseDate: string;
  statusItem: string;
  currentUserId?: number | null;
  createdAt: string;
  updatedAt: string;
  currentUser?: User;
  movements?: Movement[];
}

File: ProjetoGerenciamentoEstoque.Domain/Models/Item.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;

namespace ProjetoGerenciamentoEstoque.Domain.Models
{
    public class Item
    {
        public int Id { get; set; }
        public string Category { get; set; } 
        public string Name { get; set; }
        public string Brand { get; set; }
        public string Model { get; set; }
        public string SerialNumber { get; set; }
        public string PatrimonyNumber { get; set; }
        public string Description { get; set; }
        public DateTime PurchaseDate { get; set; }
        public string StatusItem { get; set; }
        public int? CurrentUserId { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        [JsonIgnore]
        public User? CurrentUser { get; set; }
        [JsonIgnore]
        public ICollection<Movement>? Movements { get; set; }
    }
}
